File: leetcode/001-Two-Sum/ruby/Solution.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    num_idx = {}
    nums.each_with_index do |num, i|
        if num_idx[target - num]
            return [num_idx[target - num], i]
        end
        num_idx[num] = i
    end
end

File: leetcode/001-Two-Sum/php/Solution.php
class Solution {

/**
* @param Integer[] $nums
* @param Integer $target
* @return Integer[]
*/
function twoSum($nums, $target) {
$numIdx = array();
foreach (array_values($nums) as $i => $num) {
if (array_key_exists($target - $num, $numIdx)) {
return array($numIdx[$target - $num], $i);
}
$numIdx[$num] = $i;
}
return array(-1, -1);
}
}